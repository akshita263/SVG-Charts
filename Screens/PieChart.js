import React from 'react'
import { PieChart } from 'react-native-svg-charts'
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native'
 

export default class PieChartEx extends React.PureComponent {
    render() {
        const data = [50, 10, 40, 95, -4, -24, 85, 91, 35, 53, -53, 24, 50, -20, -80]

        const randomColor = () => (
            '#' + ((Math.random() * 0xffffff) << 0).toString(16) + '000000').slice(0, 7)
 
        const pieData = data.filter((value) => value > 0).map((value, index) => ({
                value,
                svg: {
                    fill: randomColor(),
                    onPress: () => console.log('press', index),
                },
                key: `pie-${index}`,
            }))
 
        return (
            <View>
            <Text style={styles.title}>
                PIE CHART
            </Text>
            <TouchableOpacity style={styles.chart}>
            <PieChart style={{ height: 250 }} data={pieData} innerRadius="0%" padAngle={0.0174533} />
            </TouchableOpacity>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    chart:{
        margin: 20,
        borderWidth: 5,
        padding: 10,
    },
    title: {
        fontSize: 30,
        padding:20,
        alignSelf: 'center',
        fontWeight: 'bold',
    }
});

