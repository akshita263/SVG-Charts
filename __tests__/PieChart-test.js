import React from 'react';
import 'react-native';
import PieChart from '../Screens/PieChart';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<PieChart />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});