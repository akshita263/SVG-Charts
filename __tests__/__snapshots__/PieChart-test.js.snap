// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#007373",
          "height": 80,
          "marginBottom": 15,
          "padding": 13,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 37,
            "textAlign": "center",
          }
        }
      >
        PIE CHART
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderWidth": 5,
        "margin": 20,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "height": 250,
        }
      }
    >
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
